app/assets/javascripts/application-footer.js

```diff
@@ -0,0 +1,12 @@
+(function() {
+  const preloadedDataElement = document.getElementById("data-preloaded");
+
+  if (preloadedDataElement) {
```

Comment:
Check if the element exists, so this works regardless of whether `@preloaded.present?` or not.